synapse/http/client.py

```diff
@@ -176,14 +178,14 @@ def _callback() -> None:
             # valid results.
             if not has_bad_ip:
                 for i in addresses:
-                    r.addressResolved(i)
-            r.resolutionComplete()
+                    recv.addressResolved(address)
+            recv.resolutionComplete()
```

Comment:
Do these `recv`s need to be `recv()`?

Reply:
I mentioned this in 315061fcb9362a9bc4ee0f33dd05e370b5f99332, but `recv` does *not* need to be instantiated. (It actually shouldn't be, it isn't a callable necessarily!) The only reason the current code works is that the [implementation of `EndpointReceiver`](https://github.com/twisted/twisted/blob/release-21.2.0-10091/src/twisted/internet/endpoints.py#L943-L955) happens to use a class with all static methods, so instantiating it essentially has the same methods called as if it isn't instantiated.

tl;dr `recv` is meant to be an instance, not a class.